static/app/components/checkbox.tsx

```diff
@@ -1,16 +1,37 @@
 import {useEffect, useRef} from 'react';
+import styled from '@emotion/styled';
+import {useHover, usePress} from '@react-aria/interactions';
+import {mergeProps} from '@react-aria/utils';
+
+import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import {IconCheckmark, IconSubtract} from 'sentry/icons';
+import commonTheme, {FormSize} from 'sentry/utils/theme';
 
 type CheckboxProps = React.InputHTMLAttributes<HTMLInputElement>;
 
-interface Props extends Omit<CheckboxProps, 'checked'> {
+interface Props extends Omit<CheckboxProps, 'checked' | 'size'> {
   /**
    * Is the checkbox active? Supports 'indeterminate'
    */
   checked?: CheckboxProps['checked'] | 'indeterminate';
+  /**
+   *
+   */
+  size?: FormSize;
 }
 
-const Checkbox = ({checked = false, ...props}: Props) => {
+type CheckboxConfig = {borderRadius: string; box: string; icon: string};
+
+const checkboxSizeMap: Record<FormSize, CheckboxConfig> = {
+  xs: {box: '12px', icon: '10px', borderRadius: '4px'},
+  sm: {box: '16px', icon: '12px', borderRadius: '4px'},
+  md: {box: '22px', icon: '16px', borderRadius: commonTheme.borderRadius},
+};
+
+const Checkbox = ({checked = false, size = 'sm', ...props}: Props) => {
```

Comment:
Hmm not sure what the default size should be. On one hand, `md` is the most common default size for our components (buttons, inputs, etc.). On the other hand, `md` is a bit too big in most checkbox use cases (we can shrink `md`, but that'd mean both `sm` and `xs` will be even smaller).

Reply:
Was thinking the same thing. I think the problem here is that these are *form* sizes, where checkboxes should always be relatively small, but Checkbox can be used outside of forms where we might want something larger. 🤔

Reply:
There is one use of `CheckboxFancy` with size="22px": https://github.com/getsentry/sentry/blob/d0c6fd3a0f7210720c2fa4b13b04dbf9072674f3/static/app/components/sidebar/onboardingStep.tsx#L35

Reply:
I can't think of a scenario where we would want a big checkbox

Reply:
So we have these, which I think is what @malwilley was referring to. Other than this I don't think a big checkbox would ever be useful.
<img width="432" alt="Screenshot 2022-12-22 at 11 13 35 AM" src="https://user-images.githubusercontent.com/44172267/209209599-abffcb9d-8e75-4290-a9b4-a2abe8a72778.png">

Reply:
Ah god the alignment.

Also please I don't know how i feel about that inner shadow 😬

Reply:
I think the inner shadow is fine — in our UI system inner shadow = input.

Reply:
Anyway back to the main question: let's just keep `sm` as the default size, it's not ideal but it's the best configuration we have given our existing usages of checkboxes and form sizes.